SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4265,74 +4268,148 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
           /*--- Get data for this profile. ---*/
 
           vector<passivedouble> Inlet_Data = profileReader.GetDataForProfile(jMarker);
-
           unsigned short nColumns = profileReader.GetNumberOfColumnsInProfile(jMarker);
+          vector<su2double> Inlet_Data_Interpolated (nCol_InletFile);
 
-          vector<su2double> Inlet_Values(nColumns);
+          /*--- Define Inlet Values vectors before and after interpolation (if needed) ---*/
+          vector<su2double> Inlet_Values(nCol_InletFile);
+          vector<su2double> Inlet_Interpolated(nColumns);
 
-          /*--- Loop through the nodes on this marker. ---*/
+          unsigned long nRows = profileReader.GetNumberOfRowsInProfile(jMarker);
 
-          for (iVertex = 0; iVertex < geometry[MESH_0]->nVertex[iMarker]; iVertex++) {
+          /*--- Pointer to call Set and Evaluate functions. ---*/
+          vector<C1DInterpolation*> interpolator (nColumns);
```

Comment:
Is this a better way to initialize multiple pointers from the same class?

Reply:
To initialize a vector of pointers no, not really, to instantiate members of a polymorphic family there is the "Factory" design pattern.